### PR TITLE
8298241: Replace C-style casts with JavaThread::cast

### PR DESCRIPTION
**Pull request by dholmes-ora**
This is a simple cleanup RFE to get rid of old-style C casts in relation to JavaThread.

In many cases involving NULL/nullptr the cast could just be dropped. Sometimes a static cast is needed to disambiguate overloads.

A couple of reinterpret_cast are needed when doing int/ptr conversion.

static_cast is used for void* conversion.

The other changes should be self explanatory.

The changes in

src/hotspot/os_cpu/bsd_aarch64/javaThread_bsd_aarch64.cpp
src/hotspot/os_cpu/windows_aarch64/javaThread_windows_aarch64.cpp

are a bit more extensive. That code was using an alias for `this` which is completely unnecessary (and the alias creation was using the cast). This could be factored out if thought necessary.

I grep'd as best I could for the old C-style casts but can't be certain I got them all.

Testing: 
 - all builds in our tiers1-5
 - local linux x64 product, slowdebug and fastdebug
 - GHA
 - Sanity testing tiers 1-3
Thanks.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298241](https://bugs.openjdk.org/browse/JDK-8298241): Replace C-style casts with JavaThread::cast

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [5641507a](https://git.openjdk.org/jdk/pull/11682/files/5641507aca51dcf10496b3c5fcb9e01daac09609)
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [5641507a](https://git.openjdk.org/jdk/pull/11682/files/5641507aca51dcf10496b3c5fcb9e01daac09609)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [5641507a](https://git.openjdk.org/jdk/pull/11682/files/5641507aca51dcf10496b3c5fcb9e01daac09609)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11682/head:pull/11682` \
`$ git checkout pull/11682`

Update a local copy of the PR: \
`$ git checkout pull/11682` \
`$ git pull https://git.openjdk.org/jdk pull/11682/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11682`

View PR using the GUI difftool: \
`$ git pr show -t 11682`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11682.diff">https://git.openjdk.org/jdk/pull/11682.diff</a>

</details>

### CODE REVIEW

**Review by coleenp — APPROVED** [review]
Looks good!

**Comment by dholmes-ora** [comment]
Thanks @coleenp !

**Review by stefank — APPROVED** [review]

File: src/hotspot/share/runtime/frame.cpp [hunk 1]
```diff
@@ -1565,7 +1565,7 @@ void FrameValues::validate() {
       prev = fv;
     }
   }
-  // if (error) { tty->cr(); print_on((JavaThread*)nullptr, tty); }
+  // if (error) { tty->cr(); print_on(nullptr, tty); }
```
Comment (stefank) [inline src/hotspot/share/runtime/frame.cpp:1568]: This probably need the same cast as in print_frame_layout, to disambiguate the call.

**Review by sspitsyn — APPROVED** [review]
Looks good.
Thanks,
Serguei

File: src/hotspot/share/runtime/frame.cpp (hunk 1 above)
Comment (dholmes-ora) [reply src/hotspot/share/runtime/frame.cpp]: Fixed

**Comment by dholmes-ora** [comment]
Thanks for the reviews @stefank  and @sspitsyn !

/integrate

**Comment by dougxc** [comment]
Is it possible to add a test based on grep to prevent re-introduction of the unwanted C-style casts?

**Comment by dholmes-ora** [comment]
Sorry @dougxc only found your comment by accident. AFAIK we don't have any such style of tests that would try to check the source code for an anti-pattern. Our test suites don't have access to the src tree when they run.

**Comment by dougxc** [comment]
Ok, thanks. That's a shame. It seems like being able to run style checkers and other linters would be useful in cases like this.

**Pull request closed without merging**